common/v2/services/ApiService/TokenInfo/TokenInfo.ts

```diff
@@ -21,14 +22,23 @@ export default class TokenInfoService {
   }
 
   public getTokensInfo = async (contractAddresses: string[]) => {
-    const params = new URLSearchParams();
-    contractAddresses.forEach(address => params.append('contractAddress', address));
+    // Splices contract addresses for large sets of contracts to break up the calls into manageable chunks
+    const queryParams = _.chunk(contractAddresses, 20).map(contractAddressArray =>
+      createParams(contractAddressArray)
+    );
 
     try {
-      const response = await this.service.get('', { params });
-      return response.data;
+      return Promise.all(
+        queryParams.map(async (params: any) => this.service.get('', { params }))
+      ).then((res: AxiosResponse[]) => _.union(...res.map(r => r.data)));
     } catch (e) {
       throw e;
     }
   };
 }
```

Comment:
If you return the Promise in a try/catch block, the catch block never executes (it's the catch of the Promise that will deal with any errors). Either:
- return the Promise without the try/catch
- `return await Promise.all(...)` inside a try/catch

https://jakearchibald.com/2017/await-vs-return-vs-return-await/